### PR TITLE
Fix #1789: Create or replace on operation seems broken

### LINKED ISSUE #1789: Create or replace on operation seems broken
In some cases it seems that:

     client.resourceList(generated).createOrReplace()

Is not working as expected.

See:  https://github.com/dekorateio/dekorate/issues/345

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #1789

### CODE REVIEW

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/ResourceIT.java [hunk 1]
```diff
@@ -113,6 +121,51 @@ public void list() {
...
+      if (object instanceof Service) {
+        assertEquals(9998, ((Service)object).getSpec().getPorts().get(0).getTargetPort().getIntVal().intValue());
+      } else if (object instanceof ConfigMap) {
+        assertTrue(((ConfigMap)object).getData().containsKey("io"));
+        assertTrue(((ConfigMap)object).getData().containsKey("test"));
+      }
+    });
+
+    // Cleanup
+    client.resourceList(list).inNamespace(currentNamespace).deletingExisting();
```
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/ResourceIT.java]: okay, I would check and report back

[the remark this answers is not in the document]
Comment (rohanKanojia) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/ResourceIT.java]: I only checked for case when `deleteExisting` is not used. let me check the other case also

[the remark this answers is not in the document]
Comment (iocanel) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/ResourceIT.java]: BTW, I think that 

- straight delete
- using resource(i)
- using resourceList(i)

have different execution path and need to be individually tesed.

AFAIR, resourceList might not even support `deletingExisiting` but I am not even sure.

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceListTest.java [hunk 2]
```diff
@@ -86,4 +107,29 @@ public void testDelete() {
     assertFalse(deleted);
   }
 
+  @Test
+  public void testCreateOrReplaceWithoutDeleteExisting() {
+    server.expect().get().withPath("/api/v1/namespaces/ns1/services/my-service").andReturn(200 , service).once();
+    server.expect().get().withPath("/api/v1/namespaces/ns1/configmaps/my-configmap").andReturn(200, configMap).once();
+    server.expect().post().withPath("/api/v1/namespaces/ns1/services").andReturn(200, service).once();
+    server.expect().post().withPath("/api/v1/namespaces/ns1/configmaps").andReturn(200, configMap).once();
```
Comment (iocanel) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceListTest.java]: Should that be a post or a put?

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceListTest.java [hunk 3]
```diff
@@ -86,4 +107,29 @@ public void testDelete() {
...
+
+    KubernetesClient client = server.getClient();
+    KubernetesList list = new KubernetesListBuilder().withItems(service, configMap).build();
+    client.resourceList(list).inNamespace("ns1").createOrReplace();
```
Comment (iocanel) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceListTest.java]: I think that we also need to verify that the server did got everything it expected.

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**